### ISSUE TITLE
LAN server crashes on map finish on Windows

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#4714]
Multiple reports on Discord in #bugs. Can't reproduce, happens with 15.9

**Comment by def-** [comment]
Can reproduce with Wine:
```
[2022-02-15 17:35:37][chat]: *** nameless tee finished in: 0 minute(s)  0.18 second(s)
wine: Unhandled page fault on execute access to 0000000000000000 at address 0000000000000000 (thread 00c0), starting debugger...
```

**Comment by def-** [comment]
```
Unhandled exception: page fault on execute access to 0x0000000000000000 in 64-bit code (0x0000000000000000).
Register dump:
 rip:0000000000000000 rsp:0000000001cef538 rbp:0000000000a03578 eflags:00000206 (   - --  I   - -P- )
 rax:000000000055e710 rbx:0000000000a12190 rcx:0000000000a609b8 rdx:0000000000000001
 rsi:0000000001cefc40 rdi:0000000000000100  r8:ffffffffffffffff  r9:0000000000000001 r10:0000000000a80000
 r11:0000000000a80000 r12:0000000000a03570 r13:0000000001cef720 r14:0000000000a121a8 r15:0000000000516ad7
Stack dump:
0x0000000001cef538:  00000000004036a7 0000000000a12190
0x0000000001cef548:  0000000000a12190 0000000001cefc40
0x0000000001cef558:  0000000000a12190 0000000000a03570
0x0000000001cef568:  0000000000455224 0000000000a609b8
0x0000000001cef578:  0000000000000000 0000000001cef5b0
0x0000000001cef588:  0000000000516afb 0000000000a03578
0x0000000001cef598:  0000000000516ad7 3fc47ae140000000
0x0000000001cef5a8:  0000000000000000 0000000000000000
0x0000000001cef5b8:  0000000000000000 0000000000000000
0x0000000001cef5c8:  0000000000000000 0000000000000000
0x0000000001cef5d8:  0000000000000000 0000000000000000
0x0000000001cef5e8:  0000000000000000 0000000000000000
Backtrace:
=>0 0x0000000000000000 (0x0000000000a03578)
0x0000000000000000: -- no code accessible --
Modules:
Module	Address					Debug info	Name (26 modules)
PE	0000000000400000-0000000000621000	Deferred        ddnet-server
PE	0000000064940000-0000000064955000	Deferred        libwinpthread-1
PE	0000000066380000-00000000664d3000	Deferred        sqlite3
PE	000000007b000000-000000007b0d5000	Deferred        kernelbase
PE	000000007b600000-000000007b812000	Deferred        kernel32
PE	0000000170000000-000000017009a000	Deferred        ntdll
PE	00000001c69e0000-00000001c72fc000	Deferred        shell32
PE	00000001c8db0000-00000001c8e47000	Deferred        msvcrt
PE	00000001dd3f0000-00000001dd4bc000	Deferred        crypt32
PE	00000001ec2b0000-00000001ec2d5000	Deferred        ws2_32
PE	000000021a7e0000-000000021a855000	Deferred        setupapi
PE	0000000231ae0000-0000000231b62000	Deferred        rpcrt4
PE	000000023d820000-000000023da68000	Deferred        user32
PE	000000026b4c0000-000000026b53a000	Deferred        gdi32
PE	00000002d4d40000-00000002d4d56000	Deferred        bcrypt
PE	00000002de970000-00000002de9a2000	Deferred        rsaenh
PE	00000002e3540000-00000002e3591000	Deferred        shlwapi
PE	00000002e8f10000-00000002e902b000	Deferred        ole32
PE	00000002f1fa0000-00000002f1fad000	Deferred        version
PE	00000003126f0000-0000000312709000	Deferred        shcore
PE	0000000327020000-0000000327073000	Deferred        combase
PE	000000032a700000-000000032a729000	Deferred        sechost
PE	0000000330260000-000000033029f000	Deferred        advapi32
PE	0000000375610000-0000000375648000	Deferred        win32u
PE	00000003af670000-00000003af730000	Deferred        ucrtbase
PE	00000003afd00000-00000003afd1a000	Deferred        imm32
Threads:
process  tid      prio (all id:s are in hex)
00000020 (D) Z:\Users\deen\Downloads\DDNet-15.9-win64\DDNet-Server.exe
	00000024    0
	000000c0    0 <==
	000000c4    0
	000000c8    0
	000000cc    0
	000000d8    0
00000038 services.exe
	0000003c    0
	00000040    0
	0000004c    0
	0000006c    0
	00000090    0
	000000a8    0
	000000f0    0
	000000fc    0
	00000100    0
00000044 winedevice.exe
	00000048    0
	00000054    0
	00000058    0
	0000005c    0
	00000060    0
	000000b4    0
00000064 winedevice.exe
	00000068    0
	00000070    0
	00000074    0
	00000078    0
	0000007c    0
	00000080    0
	00000084    0
00000088 plugplay.exe
	0000008c    0
	00000094    0
	00000098    0
	0000009c    0
000000a0 svchost.exe
	000000a4    0
	000000ac    0
	000000b0    0
000000b8 conhost.exe
	000000bc    0
000000dc explorer.exe
	000000e0    0
	000000e4    0
	000000e8    0
000000f4 rpcss.exe
	000000f8    0
	00000104    0
	00000108    0
	0000010c    0
	00000110    0
	00000114    0
	00000118    0
System information:
    Wine build: wine-7.0
    Platform: x86_64
    Version: Windows 7
    Host system: Darwin
    Host version: 21.3.0
```

**Comment by def-** [comment]
@Zwelf Could this be related to these changes?

> [Server] Fail database queries faster [Zwelf]
> [Server] Improve database writes during shutdown and bad network [Zwelf]

**Comment by def-** [comment]
There was no SQlite upgrade yet, it will be in 16.0.

**Comment by def-** [comment]
Suspicious looking changes:
- 6feb1a0eee9de71ba3696083057573766a7d1dbc
- 3f7befdd8b2e1f0e22eb28ec4fa8df05134348e4

**Comment by def-** [comment]
Fixed with #4717
